src/LiveQuery/ParseLiveQueryServer.js

```diff
@@ -839,11 +839,25 @@ class ParseLiveQueryServer {
       return;
     }
 
+    const subscription = subscriptionInfo.subscription;
+    const className = subscription.className;
+    const trigger = getTrigger(className, 'beforeUnsubscribe', Parse.applicationId);
+    if (trigger) {
+      const auth = await this.getAuthFromClient(client, request.requestId, request.sessionToken);
+      if (auth && auth.user) {
+        request.user = auth.user;
+      }
+
+      request.sessionToken = subscriptionInfo.sessionToken;
+      request.useMasterKey = client.hasMasterKey;
+      request.installationId = client.installationId;
+
+      await runTrigger(trigger, `beforeUnsubscribe.${className}`, request, auth);
```

Comment:
What happens if user throws an error from Unsubscribe trigger? This should probably be wrapped in `try/catch` and return the error to the client

Reply:
How is that currently designed for other triggers?

Reply:
@mtrezza I think most others have try/catch. I'll add one here too